modin/core/io/file_dispatcher.py

```diff
@@ -258,16 +258,14 @@ def file_exists(cls, file_path, storage_options=None):
             Whether file exists or not.
         """
         if isinstance(file_path, str):
-            match = S3_ADDRESS_REGEX.search(file_path)
-            if match is not None:
-                if file_path[0] == "S":
-                    file_path = "{}{}".format("s", file_path[1:])
-                S3FS = import_optional_dependency(
-                    "s3fs", "Module s3fs is required to read S3FS files."
-                )
+            if is_fsspec_url(file_path) or is_url(file_path):
+                if file_path.startswith("S"):
```

Comment:
You can use `split_protocol` and check for an exact match for the protocol.

Reply:
`split_protocol` is used now for more accurate checking on supported paths.

Line `if file_path.startswith("S"):` is needed to making `S`->`s` because `fsspec` doesn't work with path starting from capital letter:

```python
In [1]: import fsspec

In [2]: path = "S3://modin-datasets/testing/multiple_csv/test_data*.csv"

In [3]: fsspec.core.url_to_fs(path)
---------------------------------------------------------------------------
ValueError                                Traceback (most recent call last)
Input In [3], in <cell line: 1>()
----> 1 fsspec.core.url_to_fs(path)

File ~/miniconda3/envs/modin/lib/python3.8/site-packages/fsspec/core.py:408, in url_to_fs(url, **kwargs)
    406 else:
    407     protocol = split_protocol(url)[0]
--> 408     cls = get_filesystem_class(protocol)
    410     options = cls._get_kwargs_from_urls(url)
    411     update_storage_options(options, kwargs)

File ~/miniconda3/envs/modin/lib/python3.8/site-packages/fsspec/registry.py:225, in get_filesystem_class(protocol)
    223 if protocol not in registry:
    224     if protocol not in known_implementations:
--> 225         raise ValueError("Protocol not known: %s" % protocol)
    226     bit = known_implementations[protocol]
    227     try:

ValueError: Protocol not known: S3
```

Reply:
Nit: what if we did something like this? Just trying to see if we can default to lowercasing the first letter so that we don't have to check for the upper case 'S'.
```python
if fsspec.core.split_protocol(file_path)[0] in (
    "s3",
    "S3",
    "https",
    "http",
):
    file_path = file_path[0].lower() + file_path[1:]
```
We can default to lowercasing the path

Reply:
It looks more clearer, I have used your suggestion + added a comment why we this do